pymysql/login_path.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+"""Funtions that read and decrypt MySQL's login path file."""
+
+from io import BytesIO, TextIOWrapper
+import os
+import struct
+
+try:
+    from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+    from cryptography.hazmat.backends import default_backend
+except ImportError:
+    raise ImportError('You must install the package "cryptography" in order '
+                      'to read the login path file.')
+
+# Buffer at the beginning of the login path file.
+UNUSED_BUFFER_LENGTH = 4
+
+# The key stored in the file.
+LOGIN_KEY_LENGTH = 20
+
+# The entire login key header.
+LOGIN_KEY_HEADER_LENGTH = UNUSED_BUFFER_LENGTH + LOGIN_KEY_LENGTH
+
+# Number of bytes used to store the length of ciphertext.
+CIPHER_STORE_LENGTH = 4
+
+
+def open_login_path_file():
+    """Open a decrypted version of the login path file."""
+    path = get_login_path_file()
+    try:
+        with open(path, 'rb') as fp:
```

Comment:
See https://github.com/mysql/mysql-server/blob/23032807537d8dd8ee4ec1c4d40f0633cd4e12f9/mysys_ssl/my_default.cc#L1496-L1520

Without it, this is insecure.